test/integration/tcp_tunneling_integration_test.cc

```diff
@@ -1799,6 +1799,63 @@ TEST_P(TcpTunnelingIntegrationTest, TestIdletimeoutWithLargeOutstandingData) {
   }
 }
 
+// Test idle timeout when connection establishment is prevented by not sending upstream response
+TEST_P(TcpTunnelingIntegrationTest, IdleTimeoutNoUpstreamConnection) {
+  // TCP tunneling with idle timeout when upstream doesn't respond behaves differently
+  // than UDP. The connection is established to upstream but the tunnel is not completed
+  // until the upstream sends response headers. The idle timeout applies to the
+  // waiting period for the upstream response.
+
+  // Configure a short idle timeout
+  config_helper_.addConfigModifier([&](envoy::config::bootstrap::v3::Bootstrap& bootstrap) -> void {
+    auto* listener = bootstrap.mutable_static_resources()->mutable_listeners(1);
+    auto* filter_chain = listener->mutable_filter_chains(0);
+    auto* config_blob = filter_chain->mutable_filters(0)->mutable_typed_config();
+
+    ASSERT_TRUE(config_blob->Is<envoy::extensions::filters::network::tcp_proxy::v3::TcpProxy>());
+    auto tcp_proxy_config =
+        MessageUtil::anyConvert<envoy::extensions::filters::network::tcp_proxy::v3::TcpProxy>(
+            *config_blob);
+
+    tcp_proxy_config.mutable_tunneling_config()->set_hostname("foo.lyft.com:80");
+    tcp_proxy_config.mutable_idle_timeout()->set_nanos(
+        std::chrono::duration_cast<std::chrono::nanoseconds>(std::chrono::milliseconds(500))
```

Comment:
To reduce test time:

```suggestion
        std::chrono::duration_cast<std::chrono::nanoseconds>(std::chrono::milliseconds(1))
```

Reply:
1 is not working, used 50.

Reply:
I think it's not enough time for upstream connection to be established.
I saw the error also with 50, I've tried 100 with multiple runs for the tests and it looks good.

Reply:
So for this to pass, you need the upstream connection to be established and the CONNECT sent, before the timeout expires? I think that's always going to be a flaky test setup without some more synchronization. Can you rework it so it isn't timing-dependent?